tests/nonlinear_failure_strategy_cut.cc

```diff
@@ -0,0 +1 @@
+#include "../benchmarks/nonlinear_channel_flow/simple_nonlinear.cc"
```

Comment:
It looks like this test problem is some sort of benchmark. So would it make sense to do some postprocessing, so that we see that our new method actually still gives us the correct solution?